python/pyspark/pandas/tests/indexes/test_base.py

```diff
@@ -1527,10 +1527,8 @@ def test_union(self):
                 almost=True,
             )
 
+            # Bug in Index.union dropping duplicated values is fixed in pandas 1.3
             if LooseVersion(pd.__version__) >= LooseVersion("1.3"):
```

Comment:
can we remove this and share one test base?

Reply:
Thanks!!

Reply:
Hmm, I'm afraid I'm against the change 6ed03e7.
It doesn't test with the actual pandas behavior we are following, so even if pandas changes the behavior in the future, we won't know it.

Reply:
Can we remove the old test cases, and only keep the tests only with pandas 1.3?

Reply:
We can leverage `unittest.skipIf`. I would prefer to avoid having duplicated test cases.

Reply:
Oh, then do we want to use

```python
    @unittest.skipIf(
        LooseVersion(pd.__version__) < LooseVersion("1.3")
    )
```

and remove all the version related logics such as `if LooseVersion(pd.__version__) >= LooseVersion("0.24"):` and `if LooseVersion(pd.__version__) >= LooseVersion("1.0.0")` ??

Reply:
Hm .. okay maybe it's a separate to topic to work on/discuss. I am fine with reverting the specific change.

Reply:
Sounds good. Just reverted to the previous fix.